clippy_utils/src/qualify_min_const_fn.rs

```diff
@@ -349,7 +349,12 @@ fn check_terminator<'tcx>(
         }
         | TerminatorKind::TailCall { func, args, fn_span: _ } => {
             let fn_ty = func.ty(body, cx.tcx);
-            if let ty::FnDef(fn_def_id, _) = *fn_ty.kind() {
+            if let ty::FnDef(fn_def_id, fn_substs) = *fn_ty.kind() {
+                let fn_def_id = match Instance::try_resolve(cx.tcx, cx.typing_env(), fn_def_id, fn_substs) {
+                    Ok(Some(fn_inst)) => fn_inst.def_id(),
+                    Ok(None) => return Err((span, format!("cannot resolve instance for {func:?}").into())),
```

Comment:
Not being able to resolve to an instance doesn't mean the function can't be const. Not sure what the syntax is but there should be some way to constrain the type to implement the trait as a const trait.

Reply:
I think the current form of const traits is recentish so I wouldn't expect many uses. Can you just add a comment about this doesn't handle where predicates?

Reply:
> I think the current form of const traits is recentish so I wouldn't expect many uses. Can you just add a comment about this doesn't handle where predicates?

I don't understand how `where` predicates could be used, and where in this situation, so I have a hard time understanding where to put this comment and what to put in it.

Reply:
Comment added, a false negative example added as well.